dev/release.sh

```diff
@@ -0,0 +1,711 @@
+            do_branch=true
+        fi
+        ;;
+    --next-beta )
+        next_method2=$(echo "x$1" | sed -e 's|^x--next-||')
+        shift
+        ;;
+    --branch )
+        do_branch=true
+        warn_branch=true
+        shift
+        ;;
+    --no-upload )
+        do_upload=false
+        shift
+        ;;
+    --no-update )
+        do_update=false
+        shift
+        ;;
+    --verbose )
+        VERBOSE=echo
+        git_quiet=
+        shift
+        ;;
+    --debug )
+        DEBUG=echo
+        do_upload=false
+        shift
+        ;;
+    --local-user )
+        shift
+        tagley=" -u $1"
+        gpgkey=" -u $1"
+        shift
+        ;;
+    --force )
+        force=true
+        shift
+        ;;
+    --help )
+        usage
+        exit 0
+        ;;
+    --manual )
+        sed -e '1,/^### BEGIN MANUAL/d' \
+            -e '/^### END MANUAL/,$d' \
+            < "$0" \
+            | pod2man \
+            | man -l -
+        exit 0
+        ;;
+    -- )
+        shift
+        break
+        ;;
+    * )
+        echo >&2 "Unknown option $1"
+        shift
+        exit 1
+        ;;
+    esac
+done
+
+$DEBUG >&2 "DEBUG: \$next_method=$next_method"
+$DEBUG >&2 "DEBUG: \$next_method2=$next_method2"
+
+$DEBUG >&2 "DEBUG: \$do_branch=$do_branch"
+
+$DEBUG >&2 "DEBUG: \$do_upload=$do_upload"
+$DEBUG >&2 "DEBUG: \$do_update=$do_update"
+$DEBUG >&2 "DEBUG: \$DEBUG=$DEBUG"
+$DEBUG >&2 "DEBUG: \$VERBOSE=$VERBOSE"
+$DEBUG >&2 "DEBUG: \$git_quiet=$git_quiet"
+
+case "$next_method+$next_method2" in
+    major+major | minor+minor )
+        # These are expected
+        ;;
+    alpha+alpha | alpha+beta | beta+beta | final+final | + | +beta )
+        # These are expected
+        ;;
+    * )
+        echo >&2 "Internal option error ($next_method, $next_method2)"
+        exit 1
+        ;;
+esac
+
+# Verbosity feed for certain commands
+VERBOSITY_FIFO=/tmp/openssl-$$.fifo
+mkfifo -m 600 $VERBOSITY_FIFO
+( cat $VERBOSITY_FIFO | while read L; do $VERBOSE "> $L"; done ) &
+exec 42>$VERBOSITY_FIFO
+trap "exec 42>&-; rm $VERBOSITY_FIFO" 0 2
+
+# Setup ##############################################################
+
+# Make sure we're in the work directory
+cd $(dirname $0)/..
+HERE=$(pwd)
+
+# Check that we have the scripts that define functions we use
+found=true
+for fn in "$HERE/dev/release-aux/release-version-fn.sh" \
+          "$HERE/dev/release-aux/release-state-fn.sh"; do
+    if ! [ -f "$fn" ]; then
+        echo >&2 "'$fn' is missing"
+        found=false
+    fi
+done
+if ! $found; then
+    exit 1
+fi
+
+# Load version functions
+. $HERE/dev/release-aux/release-version-fn.sh
+. $HERE/dev/release-aux/release-state-fn.sh
+
+# Make sure it's a branch we recognise
+orig_branch=$(git rev-parse --abbrev-ref HEAD)
+if (echo "$orig_branch" \
+        | grep -E -q \
+               -e '^master$' \
+               -e '^OpenSSL_[0-9]+_[0-9]+_[0-9]+[a-z]*-stable$' \
+               -e '^openssl_[0-9]+\.[0-9]+\.x$'); then
```

Comment:
This should be "openssl-" instead of "openssl_" (hyphen not underscore). The tar files are using a hyphen for the version name, and the "--branch" option creates it with a hyphen.

Reply:
Yeah, that was a blunder.  Thanks!

Reply:
Fixed